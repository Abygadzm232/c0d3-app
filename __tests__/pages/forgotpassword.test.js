jest.mock('@apollo/react-hooks')
import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import ForgotPassword from '../../pages/forgotpassword'
import { useMutation } from '@apollo/react-hooks'

const mockFn = jest.fn()

describe('<ForgotPassword />', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('It should render reset password form', async () => {
    useMutation.mockReturnValue([mockFn, { data: null }])
    const { container } = render(<ForgotPassword />)
    await wait(() => expect(container).toMatchSnapshot())
  })
  test('It should render password reset instructions sent', () => {
    useMutation.mockReturnValue([
      mockFn,
      { data: { success: true, token: 'faketoken' } }
    ])
    const { container } = render(<ForgotPassword />)
    expect(container).toMatchSnapshot()
  })
  test('It should render reset password form and call reqPwReset ', async () => {
    useMutation.mockReturnValue([mockFn, { data: null }])
    const { getByTestId } = render(<ForgotPassword />)
    const submitButton = getByTestId('submit')
    const userOrEmail = getByTestId('userOrEmail')

    await wait(() => {
      fireEvent.change(userOrEmail, {
        target: {
          value: 'somefakeuser'
        }
      })
      fireEvent.click(submitButton)
    })
    expect(mockFn).toBeCalledWith({
      variables: { userOrEmail: 'somefakeuser' }
    })
  })
  test('It should render invalid user/email form', () => {
    useMutation.mockReturnValue([mockFn, { data: null, error: 'fake Error' }])
    const { container } = render(<ForgotPassword />)
    expect(container).toMatchSnapshot()
  })
})
