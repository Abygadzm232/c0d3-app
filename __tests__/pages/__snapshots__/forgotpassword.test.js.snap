// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Page Should render invalid user/email form on error 1`] = `
<div>
  <nav
    class="navbar navbar navbar-expand-lg navbar-light bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <div
          class="navbar-brand text-primary fw-bold"
        >
          C0D3
          <span
            class="isAdminBadge badge"
          >
             ADMIN 
          </span>
        </div>
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="text-center navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="m-auto navbar-nav"
        >
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="/curriculum"
          >
            Curriculum
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="https://github.com/garageScript/c0d3-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            Repo
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="/docs/learn"
          >
            Learn
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="https://discord.gg/c0d3"
            rel="noopener noreferrer"
            target="_blank"
          >
            Community
          </a>
        </div>
        <div
          class="nav-buttons"
        >
          <div
            class="dropdown"
          >
            <div
              class="d-none d-lg-block"
            >
              <button
                aria-expanded="false"
                class="nav-item dropdown-toggle btn btn-primary"
                id="user_nav_toggle"
                type="button"
              >
                Fakeusername
              </button>
            </div>
            <div
              class="d-lg-none"
            >
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/lessons"
              >
                Lessons
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/users"
              >
                Users
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/alerts"
              >
                Alerts
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/profile/fakeusername"
              >
                Profile
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                <span>
                  Logout
                </span>
                <span>
                  Logout
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="row mt-5"
    >
      <div
        class="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-6 m-auto px-md-5 border-0"
      >
        <div
          class="card-body text-center pt-5 pb-5"
        >
          <h1
            class="card-title h2 fw-bold mb-5"
          >
            Username or Email does not exist
          </h1>
          <p
            class="card-text"
          />
          <a
            class="btn btn-primary btn-lg btn-block mb-3"
            href="/forgotpassword"
            role="button"
          >
            Go Back
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="mt-5 fw-light text-center pb-5 text-muted"
  >
    © Copyright 2022 GarageScript
  </footer>
</div>
`;

exports[`ForgotPassword Page Should render password reset instructions on success 1`] = `
<div>
  <nav
    class="navbar navbar navbar-expand-lg navbar-light bg-white"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <div
          class="navbar-brand text-primary fw-bold"
        >
          C0D3
          <span
            class="isAdminBadge badge"
          >
             ADMIN 
          </span>
        </div>
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="text-center navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="m-auto navbar-nav"
        >
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="/curriculum"
          >
            Curriculum
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="https://github.com/garageScript/c0d3-app"
            rel="noopener noreferrer"
            target="_blank"
          >
            Repo
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="/docs/learn"
          >
            Learn
          </a>
          <a
            class="nav-item nav-link button.path nav-link noUnderline"
            href="https://discord.gg/c0d3"
            rel="noopener noreferrer"
            target="_blank"
          >
            Community
          </a>
        </div>
        <div
          class="nav-buttons"
        >
          <div
            class="dropdown"
          >
            <div
              class="d-none d-lg-block"
            >
              <button
                aria-expanded="false"
                class="nav-item dropdown-toggle btn btn-primary"
                id="user_nav_toggle"
                type="button"
              >
                Fakeusername
              </button>
            </div>
            <div
              class="d-lg-none"
            >
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/lessons"
              >
                Lessons
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/users"
              >
                Users
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/admin/alerts"
              >
                Alerts
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="/profile/fakeusername"
              >
                Profile
              </a>
              <a
                class="nav-link dropdown-item  dropdown-item"
                data-rr-ui-dropdown-item=""
                href="#"
                role="button"
                tabindex="0"
              >
                <span>
                  Logout
                </span>
                <span>
                  Logout
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="container-md"
  >
    <div
      class="row mt-5"
    >
      <div
        class="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-6 m-auto px-md-5 border-0"
      >
        <div
          class="card-body text-center pt-5 pb-5"
        >
          <img
            alt="Green circle with a checkmark inside."
            class="mb-4"
            height="100px"
            src="/assets/curriculum/icons/checked.svg"
            width="100px"
          />
          <h1
            class="card-title h2 fw-bold mb-5"
          >
            Password reset instructions sent
          </h1>
          <p
            class="card-text"
          />
          <p>
            You will receive an email containing a link to reset your password.
          </p>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="mt-5 fw-light text-center pb-5 text-muted"
  >
    © Copyright 2022 GarageScript
  </footer>
</div>
`;
